sdk/eventhub/event-hubs/src/eventHubProducerClient.ts

```diff
@@ -425,40 +426,37 @@ export class EventHubProducerClient {
       partitionKey,
     });
 
-    let sender = this._sendersMap.get(partitionId || "");
-    if (!sender) {
-      const partitionPublishingOptions = isDefined(partitionId)
-        ? this._partitionOptions?.[partitionId]
-        : undefined;
-      sender = EventHubSender.create(this._context, {
-        enableIdempotentProducer: Boolean(this._enableIdempotentPartitions),
-        partitionId,
-        partitionPublishingOptions,
-      });
-      this._sendersMap.set(partitionId || "", sender);
-    }
-
-    const sendSpan = this._createSendSpan(options, spanContextsToLink);
-
-    try {
-      const result = await sender.send(batch, {
-        ...options,
-        partitionId,
-        partitionKey,
-        retryOptions: this._clientOptions.retryOptions,
-        tracingProperties: eventDataTracingProperties,
-      });
-      sendSpan.setStatus({ code: SpanStatusCode.OK });
-      return result;
-    } catch (error) {
-      sendSpan.setStatus({
-        code: SpanStatusCode.ERROR,
-        message: error.message,
-      });
-      throw error;
-    } finally {
-      sendSpan.end();
-    }
+    return tracingClient.withSpan(
```

Comment:
probably doesn't matter, but this `withSpan()` call is wrapping more code than the old code.

Reply:
Yeah, good call - but I had a reasoning for this! I figured moving the get-or-create logic for a sender into the callback will allow typescript to be aware that sender must exist in line 446, and I don't have to use `sender!.send` - although I definitely could if you (or anyone) feels there are downsides to wrapping this inside the callback